lightning/src/routing/router.rs

```diff
@@ -296,14 +380,23 @@ impl Writeable for Route {
 	fn write<W: crate::util::ser::Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
 		write_ver_prefix!(writer, SERIALIZATION_VERSION, MIN_SERIALIZATION_VERSION);
 		(self.paths.len() as u64).write(writer)?;
-		for hops in self.paths.iter() {
-			(hops.len() as u8).write(writer)?;
-			for hop in hops.iter() {
+		let mut blinded_tails = None;
+		for path in self.paths.iter() {
+			(path.hops.len() as u8).write(writer)?;
+			for hop in path.hops.iter() {
 				hop.write(writer)?;
+				if let Some(blinded_tail) = &path.blinded_tail {
+					if blinded_tails.is_none() { blinded_tails = Some(Vec::new()); }
+					blinded_tails.as_mut().map(|tails| tails.push(blinded_tail));
+				}
 			}
 		}
+		if blinded_tails.is_some() && blinded_tails.as_ref().unwrap().len() != self.paths.len() {
+			return Err(io::Error::new(io::ErrorKind::Other, "Missing blinded tail for path (if blinded tails are included, there must be 1 set per path)"))
```

Comment:
I'm assuming we don't want to write anything unless a blinded tail is actually present, since it's an even tlv. So we could do what you're suggesting, but that would unnecessarily allocate a vec of `None`s in most cases, and then we'd have to check that none of them are `Some` before writing, which all seems unfortunate(?).

I think it would be ok to just remove this check, since we have strong in-memory guarantees as you say?